// reactflipclock-js/dist/bundle.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);

function _extends() {
  _extends = Object.assign || function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };

  return _extends.apply(this, arguments);
}

function _inheritsLoose(subClass, superClass) {
  subClass.prototype = Object.create(superClass.prototype);
  subClass.prototype.constructor = subClass;
  subClass.__proto__ = superClass;
}

var FlipClock =
/*#__PURE__*/
function (_Component) {
  _inheritsLoose(FlipClock, _Component);

  function FlipClock() {
    var _this;

    _this = _Component.call(this) || this;
    _this.element$ = null;

    _this.setTimeWhenChanges = function (_ref) {
      var prevTime = _ref.time;
      var time = _this.props.time;

      if (time !== prevTime) {
        _this.element$.setTime(time);

        _this.element$.start();
      }
    };

    _this.flipclockRef = React.createRef();
    return _this;
  }

  var _proto = FlipClock.prototype;

  _proto.componentDidMount = function componentDidMount() {
    var flipclockRef = this.flipclockRef.current;
    this.element$ = $(flipclockRef).FlipClock(this.props.time, _extends({
      countdown: true
    }, this.props));
  };

  _proto.componentDidUpdate = function componentDidUpdate(prevProps) {
    this.setTimeWhenChanges(prevProps);
  };

  _proto.render = function render() {
    return React__default.createElement("div", {
      ref: this.flipclockRef
    });
  };

  return FlipClock;
}(React.Component);

exports.FlipClock = FlipClock;
